source/system/String.ooc

```diff
@@ -107,7 +107,11 @@ String: class {
 		match (this size) {
 			case 0 => this
 			case 1 => this toUpper()
-			case => this[0] toUpper() + this substring(1)
+			case =>
+				end := this substring(1)
+				result := this[0] toUpper() + end
+				end free()
+				result
```

Comment:
`result _buffer[0] = result[0] toUpper()` works, but it also violates the immutability of `String`s. I have to agree it looks a lot better, though.